frontend/packages/vsphere-plugin/src/components/ConnectionFormContext.tsx

```diff
@@ -0,0 +1,70 @@
+import * as React from 'react';
+import { ConnectionFormContextData } from './types';
+
+const ConnectionFormContext = React.createContext<ConnectionFormContextData | null>(null);
+
+export const ConnectionFormContextProvider: React.FC<{
+  children: React.ReactNode;
+}> = ({ children }) => {
+  const [vcenter, setVcenter] = React.useState<string>('');
+  const [username, setUsername] = React.useState<string>('');
+  const [password, setPassword] = React.useState<string>('');
+  const [datacenter, setDatacenter] = React.useState<string>('');
+  const [defaultdatastore, setDefaultdatastore] = React.useState<string>('');
+  const [folder, setFolder] = React.useState<string>('');
+  const [isDirty, setDirty] = React.useState(false);
+
+  const value = React.useMemo(
+    (): ConnectionFormContextData => ({
+      isDirty,
+      setDirty,
+
+      vcenter,
+      setVcenter: (v) => {
+        setDirty(true);
+        setVcenter(v);
+      },
+
+      username,
+      setUsername: (v) => {
+        setDirty(true);
+        setUsername(v);
+      },
+
+      password,
+      setPassword: (v) => {
+        setDirty(true);
+        setPassword(v);
+      },
+
+      datacenter,
+      setDatacenter: (v) => {
+        setDirty(true);
+        setDatacenter(v);
+      },
+
+      defaultdatastore,
+      setDefaultdatastore: (v) => {
+        setDirty(true);
+        setDefaultdatastore(v);
+      },
+
+      folder,
+      setFolder: (v) => {
+        setDirty(true);
+        setFolder(v);
+      },
+    }),
+    [datacenter, defaultdatastore, folder, isDirty, password, username, vcenter],
+  );
+
+  return <ConnectionFormContext.Provider value={value}>{children}</ConnectionFormContext.Provider>;
+};
+
+export const useConnectionFormContext = () => {
```

Comment:
why using context here ? It can be just a simple useConectionForm hook like 
```ts
export const useConnectionForm = (): ConnectionFormContextData => {
  const [vcenter, setVcenter] = React.useState<string>('');
  const [username, setUsername] = React.useState<string>('');
  const [password, setPassword] = React.useState<string>('');
  const [datacenter, setDatacenter] = React.useState<string>('');
  const [defaultdatastore, setDefaultdatastore] = React.useState<string>('');
  const [folder, setFolder] = React.useState<string>('');
  const [isDirty, setDirty] = React.useState(false);

  const callbacks = React.useMemo(() => ({
    setVcenter: (v) => {
      setDirty(true);
      setVcenter(v);
    },
    setUsername: (v) => {
      setDirty(true);
      setUsername(v);
    },
    setPassword: (v) => {
      setDirty(true);
      setPassword(v);
    },
    setDatacenter: (v) => {
      setDirty(true);
      setDatacenter(v);
    },
    setDefaultdatastore: (v) => {
      setDirty(true);
      setDefaultdatastore(v);
    },
    setFolder: (v) => {
      setDirty(true);
      setFolder(v);
    },
  }), []);

  return {
    vcenter,
    username,
    password,
    datacenter,
    defaultdatastore,
    folder,
    isDirty,
    setDirty,
    ...callbacks,
  }
};
```

The context makes it more complicated that it needs to be IMO

Reply:
Leftover, the flow should be more complex and the context would help.
When simplified to recent state, I kept the context when it has been already implemented.
I can remove it as well now if needed. On the other hand, we might add additional screens later and so will be probably implementing context again.